File: Technical/3 ClassNotes/Java/src/t6_prefix_sum/RangeSum.java
package t6_prefix_sum;

public class RangeSum {
    public static int getRangeSum(int[] prefix, int L, int R) {
        if (L == 0) return prefix[R];
        return prefix[R] - prefix[L - 1];
    }

    public static void main(String[] args) {
        int[] arr = {2, 4, 6, 8, 10};
        int n = arr.length;

        int[] prefix = new int[n];
        prefix[0] = arr[0];

        for (int i = 1; i < n; i++) {
            prefix[i] = prefix[i - 1] + arr[i];
        }

        System.out.println(getRangeSum(prefix, 1, 3)); // Output: 18
    }
}

File: Technical/3 ClassNotes/Python/t5_prefix_sum/Range_Sum.py
def get_range_sum(prefix, L, R):
    if L == 0:
        return prefix[R]
    return prefix[R] - prefix[L - 1]

arr = [2, 4, 6, 8, 10]
prefix = [0] * len(arr)
prefix[0] = arr[0]

for i in range(1, len(arr)):
    prefix[i] = prefix[i - 1] + arr[i]

print(get_range_sum(prefix, 1, 3))  # Output: 18